sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -70,7 +70,20 @@ object RDDConversions {
 object ExternalRDD {
 
   def apply[T: Encoder](rdd: RDD[T], session: SparkSession): LogicalPlan = {
-    val externalRdd = ExternalRDD(CatalystSerde.generateObjAttr[T], rdd)(session)
+    val attr = {
+      val attr = CatalystSerde.generateObjAttr[T]
```

Comment:
When `ExternalRDD.apply` calls `CatalystSerde.generateObjAttr`, `deserializer.nullable` has not been resolved yet. In other call path to `CatalystSerde.generateObjAttr`, `deserializer.nullable` may be already resolved.

I think that this problem depends on the calling context. If we fix `CatalystSerde.generateObjAttr` by using a general approach, the following approach is possible:

1. call `deserializer.nullable` in `CatalystSerde.generateObjAttr`
2. If an exception does not occurs, use a value of  `deserializer.nullable`
3. If an unresolved exception occurs, infer `nullable` based on `deserializer.dataType`

Is this OK with you instead of modifying `ExternalRDD.apply()`?